### DISCUSSION TITLE
请教如何将消息重写后重新进行处理

### DISCUSSION THREAD

**Discussion by ChrisKimZHT** [nonebot/discussions#376] [问答] [answered]
我现在有一个想实现的功能是：用自然语言向 ChatGPT 提问后，由 ChatGPT 生成对应指令再调用其他插件。

具体的思路为：消息 1 由 ChatGPT 插件处理，交给 ChatGPT 处理后，重写为包含插件 b 的指令的消息 2。再用消息 2 重新进行指令匹配，匹配到对应插件 b 后由插件 b 处理并返回消息。

不知道这个功能如何实现，想请教各位有没有什么思路。

**Comment by yanyongyu** [comment, marked as answer]
你可以用新的消息构造一个新的event并通过`handle_event`进行事件广播，或者你明确知道要调用哪个matcher的话可以直接实例化并执行run方法

**Reply by ChrisKimZHT** [reply]
感谢点拨！
